vilebot/src/main/java/com/oldterns/vilebot/handlers/user/QuotesAndFacts.java

```diff
@@ -246,7 +248,21 @@ private String dumpToPastebin( String title, String contents )
         }
         Gson gson = new Gson();
         Paste paste = gson.fromJson( response.toString(), Paste.class );
-        return paste.url;
+        return paste.url + " ( password: " + password + " )";
+    }
+
+    private String generatePassword()
+    {
+        String allowedChars = "abcdefghijklmnopqrstuvwxyz0123456789";
+        int charRange = allowedChars.length();
+        Random random = new Random();
+        StringBuilder password = new StringBuilder();
+        int passwordLength = 10;
+        for ( int i = 0; i < passwordLength; i++ )
+        {
+            password.append( allowedChars.charAt( random.nextInt( charRange ) ) );
```

Comment:
Really nice how you ended up with an algorithm that has no conditionals!

https://stackoverflow.com/questions/41107/how-to-generate-a-random-alpha-numeric-string includes additional techniques that are useful for security:

- Additional characters (lowercase and uppercase)
- Constant time operations: `StringBuilder.append` could end up resizing the `StringBuilder`. That could leak information about the size of the password.
- Operating on structures that are fixed by design and wont dynamically resize: `char[]` instead of `StringBuilder`